devices/devs.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Phoenix-RTOS
+ *
+ * phoenix-rtos-loader
+ *
+ * Device Interface
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "devs.h"
+#include "../errors.h"
+
+#define DEV_MAX_NB            8
+#define DEV_MAX_INSTANCE_NB   16
+
+/* dev_driver_t go to .bss section where is not zeroed
+ * This magic number defines whether driver was registered */
+#define DRIVER_IS_ACTIVE      0xababcdcd
+
+typedef struct {
+	int (*init)(int dn, dev_handler_t *h);
+	int active;
+} dev_driver_t;
+
+
+struct {
+	dev_handler_t devs[DEV_MAX_NB][DEV_MAX_INSTANCE_NB];
+	dev_driver_t drivers[DEV_MAX_NB];
+} devs_common;
+
+
+void devs_regDriver(int dev, int (*init)(int dn, dev_handler_t *h))
```

Comment:
`int dev` should probably be `unsigned int dev`, currently I can call `devs_regDriver(-3, NULL)` and overwrite BSS before the array :)

probably `int dn` should also be `unsigned`

Reply:
Yes, you have right. An oversight on my part.